collections/distinct.ts

```diff
@@ -16,7 +16,5 @@
  * ```
  */
 export function distinct<T>(array: readonly T[]): T[] {
-  const set = new Set(array);
-
-  return Array.from(set);
+  return Array.from(new Set(array));
```

Comment:
Nitpick: I personally prefer doing one thing per line, especially if those things are impactful, like iterating a whole array:

```suggestion
  return Array.from(
    new Set(array),
  );
```